app/src/main/java/com/keylesspalace/tusky/components/notifications/NotificationsFragment.kt

```diff
@@ -434,6 +438,27 @@ class NotificationsFragment :
         }
     }
 
+    private fun removeDuplicateOlderEntries(positionStart: Int) {
+        val dataList = adapter.snapshot().items
+
+        Log.w(TAG, "found elements in adapter "+dataList.size)
+
+        for (pos in dataList.size - 1 downTo  positionStart) {
```

Comment:
Instead of `...size - 1` you can use `dataList.lastIndex`.

I think that's better because it helps to clearly communicate that this is an indexing operation.

(I need to do that in a few other places in the code, including in code I've written recently, I only noticed this the other day).

You could also do this:

```
for ((pos, notificationViewData) in dataList.reversed().withIndex()) {
    // ...
}
```

Arguably that's even better (clearly communicates it's using the index and the value, and it's going in reverse).

Reply:
Ok, done.